lib/inferno/repositories/test_sessions.rb

```diff
@@ -26,6 +26,18 @@ def results_for_test_session(test_session_id)
           .map! { |result| results_repo.build_entity(result) }
       end
 
+      # Boolean function that returns true if some test_run is currently
+      # in progress
+      def test_running?(test_session_id)
+        session =
+          self.class::Model
+            .find(id: test_session_id)
+
+        return false if session.nil?
+
+        session.test_runs.any? { |run| run.status != 'done' }
```

Comment:
It should be possible to do this as a single db query rather than a query followed by a filter. It's really a query on test runs, so it probably makes sense to have the method live in the test runs repo. Something along the lines of:
```
def active_test_run_for_session?(test_session_id)
  self.class::Model
    .where(test_session_id: test_session_id)
    .exclude(status: 'done`)
    .count > 0
end
```